godot-core/src/meta/class_id.rs

```diff
@@ -247,31 +219,25 @@ impl ClassIdEntry {
     }
 
     fn none() -> Self {
-        Self::new(ClassIdSource::Borrowed(c""))
+        Self::new(ClassIdSource(Cow::Borrowed("")))
     }
 }
 
 // ----------------------------------------------------------------------------------------------------------------------------------------------
 
 /// `Cow`-like enum for class names, but with C strings as the borrowed variant.
-enum ClassIdSource {
-    Owned(String),
-    Borrowed(&'static CStr),
-}
+struct ClassIdSource(Cow<'static, str>);
 
 impl ClassIdSource {
     fn to_string_name(&self) -> StringName {
-        match self {
-            ClassIdSource::Owned(s) => StringName::from(s),
-            ClassIdSource::Borrowed(cstr) => StringName::__cstr(cstr),
+        match &self.0 {
+            Cow::Owned(s) => StringName::from(s),
+            Cow::Borrowed(str) => StringName::from(*str),
         }
     }
 
     fn as_cow_str(&self) -> Cow<'static, str> {
-        match self {
-            ClassIdSource::Owned(s) => Cow::Owned(s.clone()),
-            ClassIdSource::Borrowed(cstr) => Cow::Borrowed(ascii_cstr_to_str(cstr)),
-        }
+        self.0.clone()
     }
```

Comment:
I meant to entirely remove `ClassIdSource`, and use `Cow` in its place. This type is only used locally, the extra indirection/type safety doesn't add much value in my opinion :thinking: